contracts/src/structs/Subnet.sol

```diff
@@ -143,6 +143,17 @@ struct IPCAddress {
     FvmAddress rawAddress;
 }
 
+/// @notice Validator struct for gateway constructor.
+struct GenesisValidator {
+    /// The initial collateral deposited in the parent, 0 if none.
+    uint256 collateral;
```

Comment:
I don't think there is a need for a separate `weight` and `collateral` field. At the moment `weight` _is_ the collateral, it's in tokens. The voting power is just a projection of it made in Rust.

Reply:
oh, I was not aware of this, I thought it's another parameter updated in some way.

Reply:
@aakoshh this is the struct used to track validators pre-bootstrap. During this phase, we need to represent both collateral and explicit weight, if we ever want to support a hybrid model. If not, we could just represent a single weight here and have the relevant "power mutator" (stake, weight) update it.